core/remote/watcher/realtime_manager.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * @module core/remote/watcher/realtime_manager
+ * @flow
+ */
+
+const _ = require('lodash')
+const autoBind = require('auto-bind')
+
+const logger = require('../../utils/logger')
+const { MILLISECONDS, SECONDS } = require('../../utils/time')
+const { FILES_DOCTYPE } = require('../constants')
+const remoteErrors = require('../errors')
+
+/*::
+import type { CozyRealtime } from 'cozy-realtime'
+
+import type { RemoteCozy } from '../cozy'
+import type { CouchDBDoc } from '../document'
+
+export type RealtimeEvent = 'created' | 'updated' | 'deleted'
+export type RealtimeManagerOptions = {
+  events: {
+    debounceTime: number,
+    maxWaitTime: number,
+  },
+  reconnectionDelay: number,
+}
+*/
+
+const log = logger({
+  component: 'RemoteWatcher:RealtimeManager'
+})
+
+class RealtimeManager {
+  /*::
+  remoteCozy: RemoteCozy
+  eventHandler: (any) => any
+  reconnectTimeout: ?TimeoutID
+  options: RealtimeManagerOptions
+  realtimeLogsAdded: boolean
+  */
+
+  constructor(
+    remoteCozy /*: RemoteCozy */,
+    eventHandler /*: (any) => any */,
+    {
+      events: {
+        debounceTime = 200 * MILLISECONDS,
+        maxWaitTime = 5 * SECONDS
+      } = {},
+      reconnectionDelay = 10 * SECONDS
+    } /*: RealtimeManagerOptions */ = {}
+  ) {
+    this.remoteCozy = remoteCozy
+    this.options = { events: { debounceTime, maxWaitTime }, reconnectionDelay }
+    this.setEventHandler(eventHandler)
+
+    this.reconnectTimeout = null
+    this.realtimeLogsAdded = false
+
+    autoBind(this)
+  }
+
+  async start() {
+    log.debug('Subscribing to realtime events...')
+    try {
+      const realtime = await this.remoteCozy.realtime()
```

Comment:
In the future, it could also be nice to have a debounce on the server to avoid transferring many events in a short span of a time (like when a directory with many files is trashed).

Reply:
Yes it would be nice to be able to subscribe to grouped events.